terraform/groups/ecs-service/locals.tf

```diff
@@ -0,0 +1,95 @@
+  internal_api_url          = local.service_secrets["internal_api_url"]
+  oauth2_auth_uri           = local.service_secrets["oauth2_auth_uri"]
+  oauth2_redirect_uri       = local.service_secrets["oauth2_redirect_uri"]
+  account_test_url          = local.service_secrets["account_test_url"]
+  account_url               = local.service_secrets["account_url"]
+  cache_server              = local.service_secrets["cache_server"]
+  oauth2_client_id          = local.service_secrets["oauth2_client_id"]
+  oauth2_client_secret      = local.service_secrets["oauth2_client_secret"]
+  payments_api_url          = local.service_secrets["payments_api_url"]
+
+  # create a map of secret name => secret arn to pass into ecs service module
+  # using the trimprefix function to remove the prefixed path from the secret name
+  secrets_arn_map = {
+    for sec in data.aws_ssm_parameter.secret:
+      trimprefix(sec.name, "/${local.name_prefix}/") => sec.arn
+  }
+
+  service_secrets_arn_map = {
+    for sec in module.secrets.secrets:
+      trimprefix(sec.name, "/${local.service_name}-${var.environment}/") => sec.arn
+  }
+
+  # TODO: task_secrets don't seem to correspond with 'parameter_store_secrets'. What is the difference?
+  task_secrets = [
+    { "name": "COOKIE_SECRET", "valueFrom": "${local.secrets_arn_map.web-oauth2-cookie-secret}" },
+    { "name": "CHS_API_KEY", "valueFrom": "${local.service_secrets_arn_map.chs_api_key}" },
+    { "name": "CACHE_SERVER", "valueFrom": "${local.service_secrets_arn_map.cache_server}" },
+    { "name": "OAUTH2_CLIENT_ID", "valueFrom": "${local.service_secrets_arn_map.oauth2_client_id}" },  
+    { "name": "OAUTH2_CLIENT_SECRET", "valueFrom": "${local.service_secrets_arn_map.oauth2_client_secret}" },
+    { "name": "ACCOUNT_URL", "valueFrom": "${local.service_secrets_arn_map.account_url}" },
+    { "name": "INTERNAL_API_URL", "valueFrom": "${local.service_secrets_arn_map.internal.api.url}"},
+    { "name": "PAYMENTS_API_URL", "valueFrom": "${local.service_secrets_arn_map.payments_api_url}" }
+  ]
+
+  task_environment = [
+    { "name": "NODE_PORT", "value": "${local.container_port}" },
+    { "name": "API_LOCAL_URL", "valueFrom": "${var.api.local.url}"},
+    { "name": "AUTHENTICATION_MIDDLEWARE", "valueFrom": "${var.authentication.middleware}"},
+    { "name": "CACHE_SERVER", "valueFrom": "${var.cache.server}"},
+    { "name": "CDN_HOST", "valueFrom": "${var.cdn_host}"},
+    { "name": "CHS_API_KEY", "valueFrom": "${var.chs.api.key}"},
+    { "name": "CHS_URL", "valueFrom": "${var.chs_url}"},
+    { "name": "COOKIE_DOMAIN", "valueFrom": "${var.cookie_domain}"},
+    { "name": "COOKIE_EXPIRATION_IN_SECONDS", "valueFrom": "${var.expiration.in.seconds}"},
+    { "name": "COOKIE_NAME", "valueFrom": "${var.cookie_name}"},
+    { "name": "COOKIE_SECRET", "valueFrom": "${var.cookie.secret}"},
+    { "name": "COOKIE_SECURE_FLAG", "valueFrom": "${var.cookie.secret.flag}"},
```

Comment:
looks like there is a mixture of var names with underscores and full stops. Looking at the OE-web terraform we went with just underscores so i would change the var names to use underscores instead of full stops.

Reply:
Updated